File: dotnet/Multiplayer/Rooms/Actions/ActionSubmitResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms.Actions
{
    /// <summary>
    /// Response returned when actions are successfully submitted to target players.
    /// Contains the action IDs and target player information.
    /// </summary>
    public class ActionSubmitResponse : ApiResponse<ERoomActionsError>
    {
        /// <summary>
        /// The number of actions that were sent.
        /// </summary>
        public int Actions_sent { get; set; }

        /// <summary>
        /// List of IDs for the submitted actions.
        /// </summary>
        public List<string> Action_ids { get; set; } = new();

        /// <summary>
        /// List of target player IDs for the actions.
        /// </summary>
        public List<int> Target_players_ids { get; set; } = new();
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/actions/ActionSubmitResponse.java
package com.michitai.multiplayer.rooms.actions;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response returned when actions are successfully submitted to target players.
 * Contains the action IDs and target player information.
 */
public class ActionSubmitResponse extends ApiResponse {
    @JsonProperty("actions_sent")
    private int actionsSent;

    @JsonProperty("action_ids")
    private List<String> actionIds;

    @JsonProperty("target_players_ids")
    private List<Integer> targetPlayersIds;

    public int getActionsSent() {
        return actionsSent;
    }

    public void setActionsSent(int actionsSent) {
        this.actionsSent = actionsSent;
    }

    public List<String> getActionIds() {
        return actionIds;
    }

    public void setActionIds(List<String> actionIds) {
        this.actionIds = actionIds;
    }

    public List<Integer> getTargetPlayersIds() {
        return targetPlayersIds;
    }

    public void setTargetPlayersIds(List<Integer> targetPlayersIds) {
        this.targetPlayersIds = targetPlayersIds;
    }
}
